nemo_text_processing/text_normalization/hy/taggers/money.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) 2024, NVIDIA CORPORATION & AFFILIATES.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pynini
+from nemo_text_processing.text_normalization.en.graph_utils import NEMO_NOT_QUOTE, NEMO_SPACE, GraphFst, delete_space
+from nemo_text_processing.text_normalization.hy.utils import get_abs_path
+from pynini.lib import pynutil
+
+
+class MoneyFst(GraphFst):
+    """
+    Finite state transducer for classifying money, e.g.
+        "15 $" -> money { "տասնհինգ դոլար" }
+
+    Args:
+        cardinal: CardinalFst
+        decimal: DecimalFst
+    """
+
+    def __init__(self, cardinal: GraphFst, decimal: GraphFst, deterministic: bool = True):
+        super().__init__(name="money", kind="classify", deterministic=deterministic)
+        cardinal_graph = cardinal.final_graph
+        decimal_graph = decimal.final_graph
+
+        unit = pynini.string_file(get_abs_path("data/currency.tsv"))
+
+        optional_delimiter = pynini.closure(pynutil.add_weight(pynini.cross(NEMO_SPACE, ""), -100), 0, 1)
```

Comment:
Separate this over two lines. (The nesting can get muddled. 

Also do pynutil.delete

Reply:
Also note that this can induce non-deterministic behavior. May need to separate into `deterministic` and non-deterministic portions

Reply:
I separated it into two lines, thank you! But how can it cause non-deterministic behavior.

Reply:
You're relying on the `add_weight` to drive the graph. So technically it can go counter to intention if the downstream weights get really wonky. It's not likely but still a potential.

Reviewed again, no need for flag.